clinica/iotools/converters/oasis3_to_bids/oasis3_utils.py

```diff
@@ -321,5 +348,6 @@ def write_bids(
     # Perform import of imaging data next.
     for filename, metadata in scans.iterrows():
         path = Path(dataset_directory) / metadata.source_dir
-        install_bids(sourcedata_dir=path, bids_filename=to / filename)
+        if str(filename).split("_")[-1].split(".")[0] != "nan":
```

Comment:
Same comment, extract in a small function with a short docstring explaining what this is code is extracting from the filename.